L1Trigger/L1TMuonEndCap/src/TrackFinder.cc

```diff
@@ -115,47 +128,18 @@ void TrackFinder::process(
 
   // Get the geometry for TP conversions
   geometry_translator_.checkAndUpdateGeometry(iSetup);
+#ifdef PHASE_TWO_TRIGGER
+  ttgeometry_translator_.checkAndUpdateGeometry(iSetup);
+#endif
 
-  // ___________________________________________________________________________
-  // Get the conditions: firmware version, PC LUT version, pT BDT forests
-  bool new_conditions = condition_helper_.checkAndUpdateConditions(iEvent, iSetup);
-  // std::cout << "\nDo we have new conditions? Yes (1) or no (0)? -- " << new_conditions << std::endl;
-  // std::cout << "  * Before updating conditions, fw_version_ = " << fw_version_ << ", pt_lut_version_ = " << pt_lut_version_ << ", pc_lut_version_ = " << pc_lut_version_ << std::endl;
-
-  if (new_conditions) {
-
-    fw_version_ = condition_helper_.get_fw_version();
-    // No RPC or GEM hits in 2016
-    if (fw_version_ != 0 && fw_version_ < 50000) {
-      useRPC_ = false;
-      useGEM_ = false;
-    }
-
-    pt_lut_version_ = condition_helper_.get_pt_lut_version();
-    pc_lut_version_ = condition_helper_.get_pc_lut_version();
-
-    // Reload primitive conversion LUTs if necessary
-    // std::cout << "Configured with pc_lut_version_ = " << pc_lut_version_ << std::endl;
-    sector_processor_lut_.read( pc_lut_version_ );
-
-    // std::cout << "Configured with pt_lut_version_ = " << pt_lut_version_ << std::endl;
-    if ( pt_lut_version_ <= 5 ) {
-      pt_assign_engine_ = pt_assign_engine_2016_.get();
-      pt_assign_engine_->set_ptLUTVersion( pt_lut_version_ );
-    } else {
-      pt_assign_engine_ = pt_assign_engine_2017_.get();
-      pt_assign_engine_->set_ptLUTVersion( pt_lut_version_ );
-    }
-    // Reload pT LUT if necessary
-    pt_assign_engine_->load( &(condition_helper_.getForest()) );
-  }
-
-  // std::cout << "  * After changes, fw_version_ = " << fw_version_ << ", pt_lut_version_ = " << pt_lut_version_ << ", pc_lut_version_ = " << pc_lut_version_ << std::endl;
```

Comment:
How were you able to remove all the logic above?  Or did it just move somewhere?

Reply:
Khristian changed what I originally wrote, but the change was unnecessary. So I reverted them.

The condition is checked and updated automatically. The pt_assign_engine_ and sector_processor_lut_ also keep track of the versions on their own. If the versions change, they will reload on their own. Basically, instead of checking for new version here, they are now checked inside their own classes.